### PR TITLE
KEYCLOAK-18738 Improve RootAuthenticationSession description

### PR DESCRIPTION
**Pull request by martin-kanis**

### CODE REVIEW

File: server_admin/topics/threat/auth-sessions-limit.adoc [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+When a login page is opened for the first time in a web browser, {project_name} creates an object called authentication session that stores some useful information about the request.
+Whenever a new login page is opened from a different tab in the same browser, {project_name} creates a new record called authentication sub-session that is stored within the authentication session.
+Authentication requests can come from any type of clients e.g. Admin CLI. In that case a new authentication session is also created with one authentication sub-session.      
 
 NOTE: This section describes deployments that use the {jdgserver_name} provider for authentication sessions.
+
+Authentication session is internally stored as `RootAuthenticationSessionEntity`. Each `RootAuthenticationSessionEntity` can have multiple authentication sub-sessions stored within the
+`RootAuthenticationSessionEntity` as a collection of `AuthenticationSessionEntity`. {project_name} stores authentication sessions in a dedicated {jdgserver_name} cache.
```
Comment (hmlnarik) [inline server_admin/topics/threat/auth-sessions-limit.adoc]:
```suggestion
`RootAuthenticationSessionEntity` as a collection of `AuthenticationSessionEntity` objects. {project_name} stores authentication sessions in a dedicated {jdgserver_name} cache.
```

File: server_admin/topics/threat/auth-sessions-limit.adoc [hunk 2]
```diff
@@ -1,14 +1,23 @@
+
```
Comment (hmlnarik) [inline server_admin/topics/threat/auth-sessions-limit.adoc]:
```suggestion

[[_limit-authentication-sessions]]
```


Also add this anchor to sticky sessions in the installation guide:
```
[[_sticky-sessions]]
```

File: server_admin/topics/threat/auth-sessions-limit.adoc [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
 NOTE: This section describes deployments that use the {jdgserver_name} provider for authentication sessions.
+
+Authentication session is internally stored as `RootAuthenticationSessionEntity`. Each `RootAuthenticationSessionEntity` can have multiple authentication sub-sessions stored within the
+`RootAuthenticationSessionEntity` as a collection of `AuthenticationSessionEntity`. {project_name} stores authentication sessions in a dedicated {jdgserver_name} cache.
+The number of `AuthenticationSessionEntity` per `RootAuthenticationSessionEntity` contributes to the size of each cache entry. Total memory footprint of authentication session cache is determined by 
+number of stored `RootAuthenticationSessionEntity` and by number of `AuthenticationSessionEntity` within each `RootAuthenticationSessionEntity`.
+
  
-Higher memory usage and a network overload in a cluster may occur for deployments where there are many active `RootAuthenticationSession` with a lot of authentication sessions.
+Higher memory usage and a network overload in a cluster may occur for deployments where there are many active `RootAuthenticationSessionEntity` with a lot of `AuthenticationSessionEntity`.
```
Comment (hmlnarik) [inline server_admin/topics/threat/auth-sessions-limit.adoc]:
```suggestion
Higher memory usage may occur for deployments where there are many active `RootAuthenticationSessionEntity` with a lot of `AuthenticationSessionEntity`.

If the load balancer does not support or is not configured for 
link:{installguide_link}#_sticky-sessions[session stickiness], the network over in a cluster can
increase significantly. The reason for this load is that each request that lands to a node which does not own
the appropriate authentication session needs to retrieve and update the authentication session record in the owner
node which involves a separate network transmission for both the retrieval and the storage.
```

File: server_admin/topics/threat/auth-sessions-limit.adoc [hunk 4]
```diff
@@ -1,14 +1,23 @@
+
 === Limit Authentication Sessions
 
-{project_name} creates an authentication session for each authentication request within a realm. This session is internally represented by `RootAuthenticationSession`. Each `RootAuthenticationSession` can have multiple authentication sessions for a given client. In a browser authentication flow, the browser session translates to `RootAuthenticationSession` while browser tabs translate to a collection of authentication sessions within the `RootAuthenticationSession`.  
+When a login page is opened for the first time in a web browser, {project_name} creates an object called authentication session that stores some useful information about the request.
+Whenever a new login page is opened from a different tab in the same browser, {project_name} creates a new record called authentication sub-session that is stored within the authentication session.
+Authentication requests can come from any type of clients e.g. Admin CLI. In that case a new authentication session is also created with one authentication sub-session.
```
Comment (hmlnarik) [inline server_admin/topics/threat/auth-sessions-limit.adoc]:
```suggestion
Authentication requests can come from any type of clients e.g. Admin CLI. In that case a new authentication session is also created with one authentication sub-session.      
Please note that authentication sessions can be created also in other ways than via a browser flow. The text below
is applicable regardless of the source flow.

```

File: server_admin/topics/threat/auth-sessions-limit.adoc [hunk 5]
```diff
@@ -1,14 +1,23 @@
...
+Whenever a new login page is opened from a different tab in the same browser, {project_name} creates a new record called authentication sub-session that is stored within the authentication session.
+Authentication requests can come from any type of clients e.g. Admin CLI. In that case a new authentication session is also created with one authentication sub-session.      
 
 NOTE: This section describes deployments that use the {jdgserver_name} provider for authentication sessions.
+
+Authentication session is internally stored as `RootAuthenticationSessionEntity`. Each `RootAuthenticationSessionEntity` can have multiple authentication sub-sessions stored within the
+`RootAuthenticationSessionEntity` as a collection of `AuthenticationSessionEntity`. {project_name} stores authentication sessions in a dedicated {jdgserver_name} cache.
+The number of `AuthenticationSessionEntity` per `RootAuthenticationSessionEntity` contributes to the size of each cache entry. Total memory footprint of authentication session cache is determined by 
+number of stored `RootAuthenticationSessionEntity` and by number of `AuthenticationSessionEntity` within each `RootAuthenticationSessionEntity`.
+
```
Comment (hmlnarik) [inline server_admin/topics/threat/auth-sessions-limit.adoc:15]:
```suggestion

The number of maintained `RootAuthenticationSessionEntity` objects corresponds to the number 
of unfinished login flows from the browser. To keep the number of `RootAuthenticationSessionEntity` 
under control, using an advanced firewall control to limit ingress network traffic is recommended.
```

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thank you for the PR! Please see inline for comments.

**Review by hmlnarik — APPROVED** [review]

File: server_admin/topics/threat/auth-sessions-limit.adoc (hunk 2 above)
Comment (hmlnarik) [reply server_admin/topics/threat/auth-sessions-limit.adoc]: @martin-kanis Has the second part been addressed?

Comment (hmlnarik) [reply server_admin/topics/threat/auth-sessions-limit.adoc]: Ah, I see, it has :)

**Review by hmlnarik — DISMISSED** [review]
Thanks for the PR!

File: server_admin/topics/threat/auth-sessions-limit.adoc [hunk 6]
```diff
@@ -1,14 +1,30 @@
+[[_limit-authentication-sessions]]
 === Limit Authentication Sessions
 
-{project_name} creates an authentication session for each authentication request within a realm. This session is internally represented by `RootAuthenticationSession`. Each `RootAuthenticationSession` can have multiple authentication sessions for a given client. In a browser authentication flow, the browser session translates to `RootAuthenticationSession` while browser tabs translate to a collection of authentication sessions within the `RootAuthenticationSession`.  
+When a login page is opened for the first time in a web browser, {project_name} creates an object called authentication session that stores some useful information about the request.
+Whenever a new login page is opened from a different tab in the same browser, {project_name} creates a new record called authentication sub-session that is stored within the authentication session.
+Authentication requests can come from any type of clients e.g. Admin CLI. In that case a new authentication session is also created with one authentication sub-session.
```
Comment (andymunro) [inline server_admin/topics/threat/auth-sessions-limit.adoc]: Change "e.g" to ", such as the"
Add a comma after "in that case,"

File: server_admin/topics/threat/auth-sessions-limit.adoc [hunk 7]
```diff
@@ -1,14 +1,30 @@
...
+Please note that authentication sessions can be created also in other ways than via a browser flow. The text below is applicable regardless of the source flow.
```
Comment (andymunro) [inline server_admin/topics/threat/auth-sessions-limit.adoc]: Change "via" to "using"

File: server_admin/topics/threat/auth-sessions-limit.adoc [hunk 8]
```diff
@@ -1,14 +1,30 @@
...
+Authentication requests can come from any type of clients e.g. Admin CLI. In that case a new authentication session is also created with one authentication sub-session.
+Please note that authentication sessions can be created also in other ways than via a browser flow. The text below is applicable regardless of the source flow.   
 
 NOTE: This section describes deployments that use the {jdgserver_name} provider for authentication sessions.
+
+Authentication session is internally stored as `RootAuthenticationSessionEntity`. Each `RootAuthenticationSessionEntity` can have multiple authentication sub-sessions stored within the
+`RootAuthenticationSessionEntity` as a collection of `AuthenticationSessionEntity` objects. {project_name} stores authentication sessions in a dedicated {jdgserver_name} cache.
+The number of `AuthenticationSessionEntity` per `RootAuthenticationSessionEntity` contributes to the size of each cache entry. Total memory footprint of authentication session cache is determined by 
+number of stored `RootAuthenticationSessionEntity` and by number of `AuthenticationSessionEntity` within each `RootAuthenticationSessionEntity`.
```
Comment (andymunro) [inline server_admin/topics/threat/auth-sessions-limit.adoc]: add "the" before number in both cases

File: server_admin/topics/threat/auth-sessions-limit.adoc [hunk 9]
```diff
@@ -1,14 +1,30 @@
...
+
+The number of maintained `RootAuthenticationSessionEntity` objects corresponds to the number of unfinished login flows from the browser. To keep the number of `RootAuthenticationSessionEntity`
+under control, using an advanced firewall control to limit ingress network traffic is recommended.
+
  
-Higher memory usage and a network overload in a cluster may occur for deployments where there are many active `RootAuthenticationSession` with a lot of authentication sessions.
+Higher memory usage may occur for deployments where there are many active `RootAuthenticationSessionEntity` with a lot of `AuthenticationSessionEntity`.
+If the load balancer does not support or is not configured for link:{installguide_stickysessions_link}[session stickiness], the load over network in a cluster can
+increase significantly. The reason for this load is that each request that lands to a node which does not own the appropriate authentication session needs to retrieve
```
Comment (andymunro) [inline server_admin/topics/threat/auth-sessions-limit.adoc]: Change "lands to a node which" to "lands on a node that"
Always use "that" when the remainder of the sentence qualifies the first part the sentence.
Use "which" when the remainder of the sentence provides additional information, but does not qualify the first part of the sentence.

**Review by andymunro — CHANGES_REQUESTED** [review]
A few changes, Hynek.

**Comment by martin-kanis** [comment]
@andymunro Thank you Andy. I included your suggestions.

**Review by hmlnarik — APPROVED** [review]
Thank you @martin-kanis and @andymunro!

**Pull request merged**